### DISCUSSION TITLE
problem creating sparklinker

### DISCUSSION THREAD

**Discussion by yz0000** [moj-analytical-services/splink#2295] [General]
### What happens?

when creating a sparkLinker, i got this error
py4j.security.Py4JSecurityException: Method public java.lang.String org.apache.spark.sql.internal.CatalogImpl.currentCatalog() is not whitelisted on class class org.apache.spark.sql.internal.CatalogImpl
im using a databricks cluster with table access control enabled, with high concurrency mode, 13.3 runtime.

### To Reproduce

%py
a = spark.sql("select * from table_a")
b = spark.sql("select * from table_b")
settings = {
  "link_type": "link_only",
  "blocking_rules_to_generate_predictions": [
     block_on("first_name_3")
      ,block_on("last_name_3")
  ],
  "comparisons": [
   ctl.name_comparison("first_name",include_exact_match_level= False,set_to_lowercase=True),
   ctl.name_comparison("last_name",include_exact_match_level= False,set_to_lowercase=True),
    cl.array_intersect_at_sizes("col_array", [3, 1])
  ],
  }

linker = SparkLinker([a,b], settings)

### OS:

Spark 3.4.1, Scala 2.12

### Splink version:

3.9.8

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by RobinL** [comment]
See https://github.com/moj-analytical-services/splink/issues/1584#issuecomment-1718579809

Sorry, the dev team don't have access to databricks so it's quite hard for us to investigate these sorts of problems

**Comment by JonathanLaidler** [comment]
Hi, I had the same problem.
I was able to bypass it by explicitly setting the database when initialising the linker.
So something like:
linker = SparkLinker([a,b], settings, database='myDatabase')

Hope this helps.
